snowfakery/standard_plugins/Salesforce.py

```diff
@@ -7,8 +17,150 @@
 )
 from snowfakery import data_gen_exceptions as exc
 
+from snowfakery.plugins import SnowfakeryPlugin, PluginOption, PluginResult
+from snowfakery.data_gen_exceptions import (
+    DataGenNameError,
+    DataGenValueError,
+    DataGenError,
+)
+from snowfakery.standard_plugins.datasets import (
+    DatasetBase,
+    DatasetPluginBase,
+    sql_dataset,
+)
+from snowfakery.output_streams import SqlDbOutputStream
+from snowfakery.parse_recipe_yaml import TableInfo
+
+
+MAX_SALESFORCE_OFFSET = 2000  # Any way around this?
+
+# the option name that the user specifies on the CLI or API is just "orgname"
+# but using this long name internally prevents us from clashing with the
+# user's variable names.
+plugin_option_name = "snowfakery.standard_plugins.Salesforce.SalesforceQuery.orgname"
+
+
+class SalesforceConnection:
+    """Helper layer above simple_salesforce and salesforce_bulk"""
+
+    _sf = None
+
+    def __init__(self, get_orgname):
+        self.get_orgname = get_orgname
+        self.logger = getLogger(__name__)
+
+    @property
+    def sf(self):
+        """simple_salesfore client"""
+        if not self._sf:
+            self._sf, self._bulk = self._get_sf_clients(self.orgname)
+        return self._sf
+
+    def bulk(self):
+        """salesforce_bulk client"""
+        if not self._bulk:
+            self._sf, self._bulk = self._get_sf_clients(self.orgname)
+        return self._bulk
+
+    @property
+    def orgname(self):
+        """Look up the orgname in the scope"""
+        return self.get_orgname()
+
+    def query(self, *args, **kwargs):
+        """Query Salesforce through simple_salesforce"""
+        return self.sf.query(*args, **kwargs)
```

Comment:
Do you want to use query_all here? query only returns the first page of results.